test/e2e/openshift/openshift_test.go

```diff
@@ -85,6 +86,44 @@ var _ = Describe("Azure Container Cluster using the OpenShift Orchestrator", fun
 		}
 	})
 
+	It("should deploy all pods onto appropriate nodes", func() {
+		namespaces := []string{
+			"openshift-metrics",
```

Comment:
I think you can add prometheus-node-exporter- to daemonsetPrefixes and get rid of openshift-metrics here?
The test should then say:
1) if the pod is from a daemonset, don't care about its node.
2) if the pod's name starts registry-console-, don't care.
3) the pod must be on compute if and only if the pod's namespace does not start with openshift-, kube- and the namespace is not "default"